Src/beebwin.h

```diff
@@ -302,15 +345,27 @@ class BeebWin {
 	int		m_FPSTarget;
 	bool		m_JoystickCaptured;
 	JOYCAPS		m_JoystickCaps;
+	unsigned int	m_Joystick1Deadband;
```

Comment:
I suggest using an array of structs here, e.g:

```c++
struct JoystickState {
    JOYCAPS Caps;
    unsigned int Deadband;
    int PrevAxes;
    int PrevBtns;
    bool Captured;
};
```
and then
```c++
JoystickState m_JoystickState[2];
```

This would simplify the code in beebwin.cpp where you have:

```c++
int& prevBtns = (joyId == 1) ? m_Joystick2PrevBtns : m_Joystick1PrevBtns;
```

Reply:
Agreed